Wavefront.SDK.CSharp/DirectIngestion/DataIngesterService.cs

```diff
@@ -73,12 +73,22 @@ public int Report(string format, Stream inputStream)
             catch (IOException)
             {
             }
+            catch (WebException we)
+            {
+                response = (HttpWebResponse)we.Response;
+            }
```

Comment:
https://stackoverflow.com/a/4700154
If the response code is 4xx or 5xx, `request.GetResponse()` will throw WebException and we should get the response from that exception.